--- a/node_modules/@ganezdragon/tree-sitter-perl/grammar.js
+++ b/node_modules/@ganezdragon/tree-sitter-perl/grammar.js
@@ -531,7 +531,7 @@ module.exports = grammar({
       choice(
         seq(
           optional($.function_prototype),
-          optional($.function_attribute),
+          repeat($.function_attribute),
           choice(
             $.semi_colon,
             field('body', $.block),
@@ -699,6 +699,10 @@ module.exports = grammar({
 
       $.array_function,
 
+      $.array_function_add,
+      $.array_function_remove,
+      $.array_function_reverse,
+
       $.variable_declaration,
 
       $.keywords_in_hash_key,
@@ -727,6 +731,21 @@ module.exports = grammar({
       alias($.identifier, $.scalar_variable),
     ),
 
+    array_function_add: $ => prec.right(PRECEDENCE.TERM, seq(
+      choice('push', 'unshift'),
+      commaSeparated($, $._expression),
+    )),
+
+    array_function_remove: $ => prec.right(PRECEDENCE.TERM, seq(
+      choice('pop', 'shift'),
+      commaSeparated($, $._expression),
+    )),
+
+    array_function_reverse: $ => prec.right(PRECEDENCE.TERM, seq(
+      choice('reverse'),
+      commaSeparated($, $._expression),
+    )),
+
     // push_function: $ => prec.right(PRECEDENCE.TERM, seq(
     //   alias('push', $.push),
     //   with_or_without_brackets(commaSeparated($._expression)),
